### ISSUE TITLE
Proposed list of Metrics to Stabilize

### ISSUE THREAD

**Issue by rcoh** [tokio-rs/tokio#6546]
**Is your feature request related to a problem? Please describe.**
Given the impact, hassle, and perceived "risk" of compiling with `tokio_unstable`, I'd like to propose we stabilize some of the existing metrics.

**Describe the solution you'd like**
1. [Runtime::metrics](https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html#method.metrics) is stabilized
2. [RuntimeMetrics](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html) is stabilized
3. We stabilize individual metrics.

I've included the metrics below. Feel free to comment below / check any that we feel are ready for stabilization.

### Metrics we have today
- [ ] [active_tasks_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.active_tasks_count)
- [ ] [blocking_queue_depth](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.blocking_queue_depth)
- [ ] [budget_forced_yield_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.budget_forced_yield_count)
- [ ] [injection_queue_depth](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.injection_queue_depth)
- [ ] [io_driver_fd_deregistered_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.io_driver_fd_deregistered_count)
- [ ] [io_driver_fd_registered_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.io_driver_fd_registered_count)
- [ ] [io_driver_ready_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.io_driver_ready_count)
- [ ] [num_blocking_threads](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.num_blocking_threads)
- [ ] [num_idle_blocking_threads](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.num_idle_blocking_threads)
- [ ] [num_workers](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.num_workers)
- [ ] [poll_count_histogram_bucket_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.poll_count_histogram_bucket_count)
- [ ] [poll_count_histogram_bucket_range](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.poll_count_histogram_bucket_range)
- [ ] [poll_count_histogram_enabled](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.poll_count_histogram_enabled)
- [ ] [poll_count_histogram_num_buckets](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.poll_count_histogram_num_buckets)
- [ ] [remote_schedule_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.remote_schedule_count)
- [ ] [worker_local_queue_depth](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.worker_local_queue_depth)
- [ ] [worker_local_schedule_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.worker_local_schedule_count)
- [ ] [worker_mean_poll_time](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.worker_mean_poll_time)
- [ ] [worker_noop_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.worker_noop_count)
- [ ] [worker_overflow_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.worker_overflow_count)
- [ ] [worker_park_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.worker_park_count)
- [ ] [worker_poll_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.worker_poll_count)
- [ ] [worker_steal_count](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.worker_steal_count)
- [ ] [worker_steal_operations](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.worker_steal_operations)
- [ ] [worker_total_busy_duration](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.worker_total_busy_duration)

**Additional context**
https://github.com/tokio-rs/tokio/issues/4073

**Comment by Darksonn** [comment]
Please see #6114, which renames some metrics.

**Comment by rcoh** [comment]
👍🏻 , it renames `active_tasks_count` to `num_active_tasks`. I called that out in the ticket above to delay stabilization of that metric until the CR lands

**Comment by Darksonn** [comment]
As a start, do you want to submit a PR that stabilizes just the overall metrics interface and `num_workers`?

**Comment by dswij** [comment]
We'd love to see this stabilized, especially these metrics that are the most important for us:

1. `num_workers`
2. `active_tasks_count`
3. `worker_total_busy_duration`

**Comment by Darksonn** [comment]
As of today, the first metric has been stabilized in #6556.

**Comment by Owen-CH-Leung** [comment]
Do we have more metrics that are ready to be stabilised ? Right now I think only `num_alive_tasks` `injection_queue_depth` and `num_workers` are stabilised. I wonder if we can add more. Happy to file a PR

**Comment by rcoh** [comment]
yeah there are a few others in the ticket above that are ready to go. `worker_total_busy_duration` seems like one that has active customers

**Comment by Darksonn** [comment]
I don't have a good overview of which metrics are important, but I'm happy to stabilize more.

**Comment by zbraniecki** [comment]
I have a use case for `num_blocking_threads`. Can you stabilize it yet?
